src/app/shared/hooks/useTerm.ts

```diff
@@ -0,0 +1,13 @@
+import { createContext, Dispatch, SetStateAction, useContext } from 'react';
+
+import { getCurrentTerm } from '../utils/terms';
+
+const TermContext = createContext(getCurrentTerm());
+const UpdateTermContext = createContext<Dispatch<SetStateAction<string>>>(() => undefined);
```

Comment:
I don't 100% understand the reason we need a "update context".

Reply:
from the context tutorial I watched and afaik this is what you do when you have a variable context, since we wanna change the term context that's why i added it